src/main/java/net/imagej/ops/image/distancetransform/DistanceMapOp2D.java

```diff
@@ -0,0 +1,229 @@
+package net.imagej.ops.image.distancetransform;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+
+import org.scijava.plugin.Parameter;
+import org.scijava.plugin.Plugin;
+import org.scijava.thread.ThreadService;
+
+import net.imagej.ops.Contingent;
+import net.imagej.ops.Op;
+import net.imagej.ops.Ops.Image.DistanceTransform;
+import net.imagej.ops.create.img.CreateImgFromInterval;
+import net.imagej.ops.special.function.Functions;
+import net.imagej.ops.special.function.UnaryFunctionOp;
+import net.imagej.ops.special.hybrid.AbstractUnaryHybridCF;
+import net.imglib2.FinalInterval;
+import net.imglib2.RandomAccess;
+import net.imglib2.RandomAccessibleInterval;
+import net.imglib2.algorithm.labeling.ConnectedComponents.StructuringElement;
+import net.imglib2.type.BooleanType;
+import net.imglib2.type.numeric.RealType;
+
+@Plugin(type = Op.class, name = "DistanceMap2D")
+public class DistanceMapOp2D<B extends BooleanType<B>, T extends RealType<T>>
+                extends AbstractUnaryHybridCF<RandomAccessibleInterval<B>, RandomAccessibleInterval<T>> implements DistanceTransform, Contingent {
+
+        @Parameter
+        private ThreadService ts;
+
+        @Parameter(required = false)
+        private StructuringElement element;
+
+        @SuppressWarnings("rawtypes")
+        private UnaryFunctionOp<FinalInterval, RandomAccessibleInterval> createOp;
+
+        private ExecutorService es;
+
+        @Override
+        public boolean conforms() {
+                return in().numDimensions() == 2;
+        }
+
+        @Override
+        public void initialize() {
+                es = ts.getExecutorService();
+                createOp = Functions.unary(ops(), CreateImgFromInterval.class, RandomAccessibleInterval.class, new FinalInterval(in()));
```

Comment:
oh, I was misreading this. Anyway, I would force it to be `float` and not let the `CreateOp` decide.